vita3k/regmgr/src/regmgr.cpp

```diff
@@ -347,15 +392,17 @@ int32_t get_int_value(RegMgrState &regmgr, const std::string &category, const st
 
     const auto cat = fix_category(category);
     const auto &sys = regmgr.system_dreg[cat][name];
-    auto value_str = std::string(sys.begin(), sys.end());
-    if (!std::all_of(value_str.begin(), value_str.end(), [](unsigned char c) { return std::isdigit(c) || (c == '\0'); })) {
-        LOG_ERROR("Invalid value for {}{}: {}, attempt using default value!", cat, name, value_str);
-        value_str = set_default_value(regmgr, cat, name);
-        if (value_str.empty())
-            return 0;
-    }
 
-    return string_utils::stoi_def(value_str);
+    // Reverse the bytes
+    std::vector<char> reversed(4);
+    for (int i = 0; i < 4; i++)
+        reversed[i] = sys[3 - i];
+
+    // Convert to int with the reversed bytes
+    int value;
+    std::copy(reversed.begin(), reversed.end(), reinterpret_cast<char *>(&value));
+
+    return value;
```

Comment:
```suggestion
    return byte_swap(std::bit_cast<uint32_t>(sys.data()));
```
Also include the util headers for this to work.

Reply:
is see

Reply:
that is done.